TypeCobol.Analysis/Cfg/ControlFlowGraphBuilder.MultiBranchContext.cs

```diff
@@ -0,0 +1,183 @@
+﻿using System.Collections.Generic;
+using TypeCobol.Analysis.Graph;
+using TypeCobol.Compiler.Nodes;
+
+namespace TypeCobol.Analysis.Cfg
+{
+    public partial class ControlFlowGraphBuilder<D>
+    {
+        /// <summary>
+        /// The context of a Multi Branch instruction like IF or Evaluate.
+        /// </summary>
+        internal class MultiBranchContext
+        {
+            /// <summary>
+            /// Current Block before multi branches
+            /// </summary>
+            internal BasicBlockForNode CurrentBlock;
+            /// <summary>
+            /// List of multi branch blocks
+            /// </summary>
+            internal List<BasicBlockForNode> Branches;
+            /// <summary>
+            /// The Target Control Flow Graph Builder
+            /// </summary>
+            internal ControlFlowGraphBuilder<D> Builder;
+            /// <summary>
+            /// Indices in the SuccessorEdges of all blocks in Branches
+            /// </summary>
+            internal List<int> BranchIndices;
+            /// <summary>
+            /// Root Block of the multi branches
+            /// </summary>
+            internal BasicBlockForNode RootBlock;
+            /// <summary>
+            /// The Index in the SuccessorEdge of the RootBlock if any, -1 otherwise.
+            /// </summary>
+            internal int RootBlockSuccessorIndex;
+
+            /// <summary>
+            /// The instruction associated to this context.
+            /// </summary>
+            internal Node Instruction;
+
+            /// <summary>
+            /// Any contextual Data.
+            /// </summary>
+            internal object ContextualData;
```

Comment:
Always used to hold a `List<Node>` (these are the when alternatives).

Reply:
Yes, at the time I didn't know if there can be another type of contextual data.

Reply:
done